chemicaltoolbox/xchem-deep/xchem_deep_score.py

```diff
@@ -0,0 +1,192 @@
+# Create dir containing ligands.sdf and protein.pdb
+# Enter docker container like this:
+#   docker run -it --rm --gpus all -v $PWD:/root/train/fragalysis_test_files/work:Z informaticsmatters/deep-app-ubuntu-1604:latest bash
+#
+# Now inside the container run like this:
+#   rm -rf work/* && python3 -m  pipelines_obabel.gnina.xchem_deep_score -i ligands.sdf -r protein.pdb -w work
+#
+# If testing with no GPU you can use the --mock option to generate random scores
+#
+# Start container for testing like this:
+#    docker run -it --rm -v $PWD:$PWD:Z -w $PWD informaticsmatters/deep-app-ubuntu-1604:latest bash
+# Inside container test like this:
+#   mkdir /tmp/work
+#   cd chemicaltoolbox/xchem-deep
+#   rm -rf /tmp/work/* && python3 xchem_deep_score.py -i test-data/ligands.sdf -r test-data/receptor.pdb -w /tmp/work --mock
+#
+
+import argparse, os, re, sys
+import random
+from openbabel import pybel
+
+types_file_name = 'inputs.types'
+types_file_name = 'inputs.types'
+predict_file_name = 'predictions.txt'
+work_dir = '.'
+inputs_protein = []
+inputs_ligands = []
+
+def log(*args, **kwargs):
+    """Log output to STDERR
+    """
+    print(*args, file=sys.stderr, **kwargs)
+
+def write_inputs(protein_file, ligands_file):
+    global types_file_name
+    global work_dir
+    global inputs_protein
+    global inputs_ligands
+
+    ligands_path = "{0}{1}ligands".format(work_dir, os.path.sep)
+    log("Writing ligands to", ligands_path)
+    os.mkdir(ligands_path)
+    cmd1 = "gninatyper {0} {1}{2}ligands{2}ligand".format(ligands_file, work_dir, os.path.sep)
+    log('CMD:', cmd1)
+    exit_code = os.system(cmd1)
```

Comment:
this should use subprocess ... os.system() should not be used, isn't it?

Reply:
Done: 0b618d09fbce0225f991fe1bf8ecf637496cd5d2